numpy/lib/format.py

```diff
@@ -917,11 +922,13 @@ def open_memmap(filename, mode='r+', dtype=None, shape=None,
         )
         # If we got here, then it should be safe to create the file.
         with open(os_fspath(filename), mode+'b') as fp:
+            fp.seek(offset)
             _write_array_header(fp, d, version)
-            offset = fp.tell()
+            offset += fp.tell()
```

Comment:
`_write_array_header` moves the file pointer? so this should be `offset = fp.tell()`

Reply:
Yes. `_write_array_header` and `read_magic` (line :932) move the file pointer and the offset from `fp.tell()` is used by `numpy.memmap` (line :952) to write the array data starting from that position.

Reply:
I am not following, I agree it looks wrong.  But please just add that test and that will clarify it one way or the other.

Just add some test that saves to a file after writing `N` bytes first?  That `N` could be parametrized for `N=0, something` also.